android/properties.gradle.kts

```diff
@@ -41,3 +41,7 @@ val keystoreProperties = readProperties((project.properties["keystorePropertiesF
 requiredKeystoreProperties.forEach { property ->
   project.extra.set(property, keystoreProperties.getProperty(property,"sample_" + property))
 }
+
+// set APP_ID property
+val appIdProperty = "APP_ID"
```

Comment:
The appIdProperty variable holds the string that we will use as the key in the project.extra map. 

The flow is 
1. Extract APP_ID from localProperties using localProperties.get()
      if APP_ID was never added into the localProperties file as a value, it would return null so the """"""""" is the fallback. i.e if null set as empty string

2. After setting in project.extra, it is added as a buildConfigField  in the build.gradle section called defaultConfig{} that is used to autogenerate an object called BuildConfig at compile time
  
3. From there BuildConfig.APP_ID can be accessed from within the app. If we defined a value for APP_ID we get it, if we didn't we get an empty string as per the `line below` in your question

Reply:
Convert `val appIdProperty = "APP_ID"` to a constant and declare it at the top of the file to improve the readability. 

```kotlin 
const val APP_ID_PROPERTY = "APP_ID"
```

Reply:
See my suggestion in the comment below, this property will no longer be necessary if we include `OPENSRP_APP_ID` to the list of `requiredFhirProperties`